ocaml/xapi/bios_strings.ml

```diff
@@ -23,6 +23,50 @@ let remove_invisible str =
   let str = String.concat "\n" l in
   String.fold_left (fun s c -> if c >= ' ' && c <= '~' then s ^ (String.of_char c) else s) "" str
 
+(* A single record from the output of dmidecode,
+ * without its type, handle nor size *)
+type record = {name: string; values: (string * string) list}
+
+(* Parser module for the output of command dmidecode -q *)
+module P = struct
+  open Angstrom
+  let is_eol =   function | '\n' -> true | _ -> false
+  let is_tab =   function | '\t' -> true | _ -> false
+  let is_space = function | ' '  -> true | _ -> false
+  let is_sep =   function | ':'  -> true | _ -> false
```

Comment:
I think it would be good to include as a comment a typical entry that the code is supposed to parse.